### DISCUSSION TITLE
Sonoff Wall Switch T3 not toggling

### DISCUSSION THREAD

**Discussion by BigFan00777** [arendst/Tasmota#14244] [Q&A] [answered]
Hi all
I have a Sonoff Wall Switch - flashed with the latest Tasmota firmware - 

https://www.banggood.com/SONOFF-T3-EU-or-US-or-UK-AC-100-240V-1-or-2-or-3-Gang-TX-Series-WIFI-Wall-Switch-433Mhz-RF-Smart-Wall-Touch-Light-Switch-For-Smart-Home-Work-With-Alexa-Google-Home-p-1470799.html?utm_source=googleshopping&utm_medium=cpc_organic&gmcCountry=ZA&utm_content=minha&utm_campaign=minha-za-en-pc&currency=ZAR&cur_warehouse=CN&createTmp=1&ID=6278522&utm_source=googleshopping&utm_medium=cpc_bgs&utm_content=lynna&utm_campaign=lynna-ssc-za-en-all-1021-2021-11sale-21&ad_id=554760544979&gclid=Cj0KCQiAt8WOBhDbARIsANQLp94jO5wHHCVCLSC94yCVLuh0TjfiivSlrFand8wyrUBjhhMAQKE-APgaAmaZEALw_wcB

I want the relay to operate only when I push the LED Capacitive Switch and when I release the Switch, the relay should release. I have set SwitchOptions to 0, 7, 11, with no difference.  If I push the LED Capacitor Touch button/switch the relays switches on and stays on until I touch the switch again. It seems like the switch is in "lock" mode?

Any ideas?

**Comment by barbudor** [comment, marked as answer]
Default template is defining a Button, not a Switch so Switchmode has no effect
Did you changed the template?

Good reading for tonight to get the difference between Button and Switch
https://tasmota.github.io/docs/Buttons-and-Switches/

**Reply by BigFan00777** [reply]
Thank you for your quick reply. I changed to Switches as you suggested with no luck... If I touch the button, the relay switches on. I have to touch it again for the relay to switch off :-(
It's like Switchmode does nothing?

![image](https://user-images.githubusercontent.com/77881184/147883144-a5926067-daf0-479f-897e-6d15581701bf.png)
These switches have a link to make it latchable or non-latchable
Do I perhaps need to modify the cct board on the Sonoff device to make the switches non-locking?

**Reply by barbudor** [reply]
I don't believe the "latching" is done by the hardware (could be but I've not seen that on similar wall switches)
Please provide the output of command `Backlog template; module; switchmode` 
And a copy of the webconsole log when you are touching the button

**Reply by BigFan00777** [reply]
Hi Barbudor
Thank you for your assistance. As requested:

07:21:15.891 CMD: backlog template
07:21:15.966 MQT: stat/Tasmota_3Gang_Light_Sw/RESULT = {"NAME":"Generic","GPIO":[160,1,1,1,226,225,161,162,224,320,0,0,0,0],"FLAG":0,"BASE":30}
07:21:33.782 CMD: module
07:21:33.790 MQT: stat/Tasmota_3Gang_Light_Sw/RESULT = {"Module":{"30":"Sonoff T1 3CH"}}
07:21:39.291 CMD: switchmode
07:21:39.298 MQT: stat/Tasmota_3Gang_Light_Sw/RESULT = {"SwitchMode1":11}
07:22:19.964 MQT: stat/Tasmota_3Gang_Light_Sw/RESULT = {"POWER1":"ON"}
07:22:19.969 MQT: stat/Tasmota_3Gang_Light_Sw/POWER1 = ON
07:22:23.023 MQT: stat/Tasmota_3Gang_Light_Sw/RESULT = {"POWER1":"OFF"}
07:22:23.026 MQT: stat/Tasmota_3Gang_Light_Sw/POWER1 = OFF
07:22:43.156 MQT: stat/Tasmota_3Gang_Light_Sw/RESULT = {"POWER1":"ON"}
07:22:43.160 MQT: stat/Tasmota_3Gang_Light_Sw/POWER1 = ON
07:23:04.707 MQT: stat/Tasmota_3Gang_Light_Sw/RESULT = {"POWER1":"OFF"}
07:23:04.711 MQT: stat/Tasmota_3Gang_Light_Sw/POWER1 = OFF

**Reply by sfromis** [reply]
When you configure Module 30, you are not using the template where you changed to gpio0 being Switch. As a result, the default pin definition of gpio0 being a Button inherent in your module selection still applies, and thus SwitchOption makes no difference in this setup. Selecting Module 0 will activate the template.

**Reply by BigFan00777** [reply]
Thank you so much Sfromis! Setting the Module to 0 solved the problem :-)

**Reply by Adefx** [reply]
I was looking so deperatly for this solution ! 
Thanks a lot
